wcfsetup/install/files/lib/core.functions.php

```diff
@@ -140,6 +140,40 @@ function logThrowable(\Throwable $e, &$logFile = null): string {
 			return str_replace("\n", ' ', $item);
 		};
 		
+		$getExtraInformation = function (\Throwable $e) {
+			$extraInformation = [];
+			$contextLineCount = 5;
+			
+			if ($e instanceof IExtraInformationException) {
+				$extraInformation = $e->getExtraInformation();
+			}
+			
+			foreach ($e->getTrace() as $traceEntry) {
+				if (isset($traceEntry['file']) && preg_match('~/templates/compiled/.+\.php$~', $traceEntry['file'])) {
+					$lines = file($traceEntry['file']);
```

Comment:
This is unsafe for larger templates. Use something like the following to handle the file in a streaming fashion:

```
$f = \fopen($traceEntry['file'], 'r');
for ($line = 0; $line < $startLine; $line++) {
  if (\substr(\fgets($f, 1024), -1) !== "\n") {
    break 2; // We don't want to handle a file where lines exceed 1024 Bytes.
  }
}
```